modular_skyrat/modules/sec_haul/code/melee/baton.dm

```diff
@@ -0,0 +1,75 @@
+// Security officer: Surprisingly, the only change is that it takes 5 hits to crit.
+// Riot armor: Resist first two hits, knockdowno n third and later, crit on 9th.
+// Contractor MODsuit: Resist first two, knockdown on third, crit on 7th.
+// Fully kitted-out miner with full ash drake armor, talisman, gas mask, etc: Resist first six, crit on 13th.
+/obj/item/melee/baton/security
+	swings_to_knockdown = 2
+	armor_for_extra_swing_needed_for_knockdown = 0.20
+
+/// Very slow high-damage attacks that instantly knock down anyone without an armor value of .15, which is surprisingly difficult to get as, say, an assistant.
+/// Compared to the stun baton, the contractor baton crits faster, knocks down faster, but is less flexible given it's cooldown. Still objectively better, but
+/// not egregiously. You could absolutely win a baton duel against this, you just have to be good.
+// No armor: Knockdown, then crit. Security officer: Resist first knockdown, weakly crawl up on second hit, crit on third.
+// Riot armor: Resist first knockdown, knockdown on second, weakly crawl up on third hit, crit on fourth.
+/obj/item/melee/baton/telescopic/contractor_baton
+	swings_to_knockdown = 1
+	stamina_damage = 85
+	cooldown = 4 SECONDS
+	knockdown_time = 2.5 SECONDS
+	armor_for_extra_swing_needed_for_knockdown = 0.15
+
+// Override to make batons respect armor and delay knockdown
+/obj/item/melee/baton/baton_effect(mob/living/target, mob/living/user, modifiers, stun_override)
+	var/trait_check = HAS_TRAIT(target, TRAIT_BATON_RESISTANCE)
+	if(iscyborg(target))
+		if(!affect_cyborg)
+			return FALSE
+		target.flash_act(affect_silicon = TRUE)
+		target.Paralyze((isnull(stun_override) ? stun_time_cyborg : stun_override) * (trait_check ? 0.1 : 1))
+		additional_effects_cyborg(target, user)
+	else
+		if(ishuman(target))
+			var/mob/living/carbon/human/human_target = target
+			if(prob(force_say_chance))
+				human_target.force_say()
+		target.apply_damage(stamina_damage, STAMINA, blocked = target.run_armor_check(attack_flag = MELEE, armour_penetration = src.armour_penetration, weak_against_armour = src.weak_against_armour))
+		if(!trait_check)
+			var/stamina_damage_threshold_for_knockdown = ((stamina_damage * swings_to_knockdown) * TRANSLATE_EXTRA_SWING_ARMOR(armor_for_extra_swing_needed_for_knockdown))
+			if (target.staminaloss > stamina_damage_threshold_for_knockdown)
+				target.Knockdown((isnull(stun_override) ? knockdown_time : stun_override))
+		additional_effects_non_cyborg(target, user)
+	return TRUE
+
+/obj/item/melee/baton/security/apply_stun_effect_end(mob/living/target)
+	var/stamina_damage_threshold_for_knockdown = ((stamina_damage * swings_to_knockdown) * TRANSLATE_EXTRA_SWING_ARMOR(armor_for_extra_swing_needed_for_knockdown))
+	if (target.staminaloss > stamina_damage_threshold_for_knockdown)
+		var/trait_check = HAS_TRAIT(target, TRAIT_BATON_RESISTANCE) //var since we check it in out to_chat as well as determine stun duration
```

Comment:
helps with readability of the code, you dont instantly understand what this is supposed to do

Reply:
Isn't that the purpose of code comments?

Reply:
Yes, but you can also do it this way. The cost for putting a variable here is so absolutely minute that I don't think it really matters

Reply:
Actually to be specific, it's generally a better idea to just have the code describe itself rather than use comments

Reply:
> Actually to be specific, it's generally a better idea to just have the code describe itself rather than use comments

While not everything needs to have code comments, at the same time, it is beneficial to write code comments to express the intentions and thoughts behind why something does what it does, especially when it isn't clear cut.

The code here is straightfoward enough that I don't think you need this variable (or really a comment either)

Reply:
ehhh sure